contrib/kafka/Readme.md

```diff
@@ -0,0 +1,103 @@
+# Knative KafkaEventSource
+
+This deploys an Operator that can instantiate Knative Event Sources which subscribe to a Kafka Topic.
+
+## EventSource
+
+The image that receives messages from Kafka (Receive Adaptor) is available at `docker.io/sjwoodman/kafkaeventsource`.
+To build it from source edit the `eventsource/Makefile` with suitable tags for your DockerHub organisation and perform the following commands.
+
+```bash
+make docker_build docker_push_latest
+```
+
+You will need to edit the _Operator_ with the location of the newly built image.
+
+```bash
+sed -i 's|sjwoodman|DH_ORG|g' kafkaeventsource-operator/deploy/operator.yaml
+sed -i 's|sjwoodman|DH_ORG|g' kafkaeventsource-operator/pkg/controller/kafkaeventsource
+```
+
+## Usage
+
+The sample connects to a [Strimzi](http://strimzi.io/quickstarts/okd/) Kafka Broker running inside OpenShift.
+
+1. Setup [Knative Eventing](https://github.com/knative/docs/tree/master/eventing).
+
+1. Install the [in-memory `ClusterChannelProvisioner`](https://github.com/knative/eventing/tree/master/config/provisioners/in-memory-channel).
+    - Note that you can skip this if you choose to use a different type of `Channel`. If so, you will need to modify `01-channel.yaml` before deploying it.
+
+1. Install the KafkaEventSource CRD
+
+    ```bash
+    kubectl create -f deploy/crds/sources_v1alpha1_kafkaeventsource_crd.yaml
```

Comment:
May want to specify that the user needs to be in the `kafkaeventsource-operator` directory before running the scripts in the `deploy` directory.